examples/message_hash.py

```diff
@@ -0,0 +1,242 @@
+import asyncio
+import functools
+import logging
+import os
+import traceback
+
+from crypto_cpp_py.cpp_bindings import cpp_hash, get_cpp_lib_file
+from typing import cast, Sequence, List, Union
+from web3.auto import w3
+
+from starknet_py.cairo.felt import encode_shortstring
+from starknet_py.common import int_from_bytes
+from starknet_py.hash.selector import get_selector_from_name
+from starkware.cairo.lang.vm.crypto import pedersen_hash as default_hash
+
+from utils import generate_paradex_account, get_paradex_config
+
+# This is a very stripped down version of message hashing
+# Added notes around the code to explain what's going on
+
+# These types are specific to each message
+# This one is specifically for onboarding
+types = {
+    "StarkNetDomain": [
+        {"name": "name", "type": "felt"},
+        {"name": "chainId", "type": "felt"},
+        {"name": "version", "type": "felt"},
+    ],
+    "Constant": [
+        {"name": "action", "type": "felt"},
+    ],
+}
```

Comment:
I see in paradex/amm/api_client_utils.py:auth_message() this code:
```
"types": {
            "StarkNetDomain": [
                {"name": "name", "type": "felt"},
                {"name": "chainId", "type": "felt"},
                {"name": "version", "type": "felt"},
            ],
            "Request": [
                {"name": "method", "type": "felt"},
                {"name": "path", "type": "felt"},
                {"name": "body", "type": "felt"},
                {"name": "timestamp", "type": "felt"},
                {"name": "expiration", "type": "felt"},
            ],
        },
        ```
        but no `Constant` field.
        Both will work I guess ?

Reply:
Both are for different message payloads, see `examples/message_hash.md` for context.